src/deadline/client/ui/dialogs/submit_job_to_deadline_dialog.py

```diff
@@ -409,6 +410,19 @@ def on_submit(self):
                 settings.submitter_name, settings.name
             )
 
+            # First filter the queue parameters to exclude any from the job template,
+            # then extend it with the job template parameters.
+            job_parameter_names = {param["name"] for param in settings.parameters}
```

Comment:
It's unclear to me from the PR description why this is needed. Is this for the same requirement of printing out additional information or is this a separate issue being fixed?

Reply:
This block is copied from here: https://github.com/aws-deadline/deadline-cloud/blob/b5de504918cc6420d80d5206f9332cb8880471a5/src/deadline/client/ui/job_bundle_submitter.py#L101-L111

I looked at moving this code block into the core submitter class instead of as part of a callback the submitter caller provides, but it appears that the submitter class is a public interface that other things depend on, so I opted not to clean it up. Copying this code block seemed like a practical path forward.

Reply:
I guess the part I'm curious about is what use case was failing here and is it now confirmed that it's working because of this change? I want to make sure nothing else is missing that would be needed in here like the merging of queue and job parameters etc from that other file.

Reply:
I took a closer look. In the linked code, `parameter_values` contains the merged values and `parameters` contains the merged parameter definitions. I copied over the value merging logic which is sufficient. The linked code has a call to `merge_queue_job_parameters` but (confusingly) it's used for merging definitions for the UI, not the values. Since we're not storing the parameter definitions, we don't need to merge or handle them.

Reply:
Okay, so just the values being merged and that's all we need. Thanks for double checking.